packages/katello/katello/hostname-change.rb

```diff
@@ -87,27 +89,40 @@ def precheck
       # Taken from https://www.safaribooksonline.com/library/view/regular-expressions-cookbook/9781449327453/ch08s15.html
       hostname_regex = /^([a-z0-9]+(-[a-z0-9]+)*\.)+[a-z]{2,}$/
       unless hostname_regex === @new_hostname
-        self.fail_with_message("#{@new_hostname} is not a valid fully qualified domain name, please use a valid FQDN and try again. " \
+        self.fail_with_message("#{@new_hostname} is not a valid fully qualified domain name. Please use a valid FQDN and try again. " \
                           "No changes have been made to your system.");
       end
 
       unless @foreman_proxy_content
         STDOUT.puts "\nChecking overall health of server"
-        self.run_cmd("hammer ping", [0], "There is a problem with the server, please check 'hammer ping'")
+        self.run_cmd("hammer ping", [0], "There is a problem with the server; please check 'hammer ping'")
         STDOUT.puts "\nChecking credentials"
-        self.hammer_cmd("capsule list", [0], "There is a problem with the credentials, please retry")
+        self.hammer_cmd("capsule list", [0], "There is a problem with the credentials; please retry")
       end
 
-      if @options[:confirm]
-        response = true
-      else
-        STDOUT.print(self.warning_message)
-        response = agree('Proceed with changing your hostname? [y/n]')
+      if should_update_dns?
+        begin
+          STDOUT.puts "\nRunning test DNS query"
+          query_dns = Resolv::DNS.new(nameserver: ['127.0.0.1'], search: [], ndots: 1)
```

Comment:
In the answers file you can find the actual nameserver used by nsupdate (`answers['foreman_proxy']['dns_server']`). There's also a `query_dns_server` helper that duplicates the constructor. Should there be a `dns_resolver` helper?